### PR TITLE
GaKa00/issue42

### LINKED ISSUE #42: get products by stock amount endpoints
```
  //TODO get products by stock amount endpoints @GaKa00 
```

https://github.com/JohanTran02/IMS/blob/5e1d7eb04b6b0a2fa6c73c21d922b8e4b2f4a3ce/backend/resources/product/products.controller.ts#L203

### PR DESCRIPTION
**Pull request by GaKa00**
Fixes #42 and bool error #43

### CODE REVIEW

File: backend/resources/product/products.controller.ts [hunk 1]
```diff
@@ -199,4 +199,65 @@ export async function getManufacturers(req, res) {
...
+        $group: {
+          _id: "$product.name",
+         name: { $first: "$product.name" },
+          amountInStock: { $first: "$amountInStock" },
+        },
+      },
+    
+      bool ? {
+        $sort: { amountInStock: -1  }
+      } : { $sort: { amountInStock: 1 } }
```
Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:221]:
```suggestion
    $sort: { amountInStock: bool  }
```

File: backend/resources/product/products.controller.ts [hunk 2]
```diff
@@ -199,4 +199,65 @@ export async function getManufacturers(req, res) {
...
+      category:Result.category,
+      name: Result.name,
+      
+    }));
+
+    res.status(200).json(FilteredValue);
+  } catch (err) {
+    res.status(500).json({ message: "Server Error", error: err.message });
+  }
+}
```
Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:263]:
```suggestion
     { $sort: { amountInStock: -1  }}
```

Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:263]:
```suggestion
     { $sort: { amountInStock: -1  }}
```

Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:263]:
```suggestion
     { $sort: { amountInStock: -1  }}
```

Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:263]:
```suggestion
     { $sort: { amountInStock: -1  }}
```

Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:263]:
```suggestion
     { $sort: { amountInStock: -1  }}
```

Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:263]:
```suggestion
     { $sort: { amountInStock: -1  }}
```

Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:263]:
```suggestion
     { $sort: { amountInStock: -1  }}
```

Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:263]:
```suggestion
     { $sort: { amountInStock: -1  }}
```

Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:263]:
```suggestion     
        { $sort: { amountInStock: -1  }}
```

Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:263]:
```suggestion
    
      
             { $sort: { amountInStock: -1  }}
      
```

Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:263]:
```suggestion
            { $sort: { amountInStock: -1  }}
```

Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:263]:
```suggestion
        { $sort: { amountInStock: -1  }}
    
```

Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:263]:
```suggestion
      bool ? {
        $sort: { amountInStock: -1  }
      } : { $sort: { amountInStock: 1 } }
```     { $sort: { amountInStock: -1  }}

Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:263]:
```suggestion
{ $sort: { amountInStock: -1  }}
```

Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:263]:
```suggestion
            { $sort: { amountInStock: -1  }}
```

Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:263]:
```suggestion
     { $sort: { amountInStock: -1  }}
    ]);
```

File: backend/resources/product/products.controller.ts (hunk 1 above)
Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:221]:
```suggestion
     { $sort: { amountInStock: -1  }}
```

File: backend/resources/product/products.controller.ts [hunk 3]
```diff
@@ -199,4 +199,65 @@ export async function getManufacturers(req, res) {
...
+
+    const bool: 1 | -1 = parseInt(req.query.sort) === 1 ? 1 : -1;
+
+  try {
+    const result = await Product.aggregate([
+      {
+        $group: {
+          _id: "$product.name",
+         name: { $first: "$product.name" },
+          amountInStock: { $first: "$amountInStock" },
```
Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:215]:
```suggestion
     test
```

File: backend/resources/product/products.controller.ts (hunk 1 above)
Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:221]:
```suggestion
     { $sort: { amountInStock: -1  }}
```

File: backend/resources/product/products.controller.ts (hunk 2 above)
Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:263]:
```suggestion
      {
        $group: {
          _id: "$_id",
          name: { $first: "$name" },
          amountInStock: { $first: "$amountInStock" },
        },
      },
```

Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:263]:
```suggestion
      {
        $group: {
          _id: "$name",
          name: { $first: "$name" },
          amountInStock: { $first: "$amountInStock" },
        },
      },
```

Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:263]:
```suggestion
      {
        $group: {
          _id: "$name",
          name: { $first: "$name" },
          amountInStock: { $first: "$amountInStock" },
        },
      },
```

Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:263]:
```suggestion
      {
        $group: {
          _id: "$name",
          name: { $first: "$name" },
          amountInStock: { $first: "$amountInStock" },
        },
      },
```

Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:263]:
```suggestion
      {
        $group: {
          _id: "$name",
          name: { $first: "$name" },
          amountInStock: { $first: "$amountInStock" },
        },
      },
```

Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:263]:
```suggestion
      {
        $group: {
          _id: "$name",
          name: { $first: "$name" },
          amountInStock: { $first: "$amountInStock" },
        },
      },
```

Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:263]:
```suggestion
      {
        $group: {
          _id: "$name",
          name: { $first: "$name" },
          amountInStock: { $first: "$amountInStock" },
        },
      },
```

Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:263]:
```suggestion
      {
        $group: {
          _id: "$name",
          name: { $first: "$name" },
          amountInStock: { $first: "$amountInStock" },
        },
      },
```

Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:263]:
```suggestion
        $group: {
          _id: "$name",
          name: { $first: "$name" },
          amountInStock: { $first: "$amountInStock" },
        },
```

Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:263]:
```suggestion
      {
        $group: {
          _id: "$name",
          name: { $first: "$name" },
          amountInStock: { $first: "$amountInStock" },
        },
      },
```

Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:263]:
```suggestion
        $group: {
          _id: "$name",
          name: { $first: "$name" },
          amountInStock: { $first: "$amountInStock" },
        },
```

Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:263]:
```suggestion
    $group: {
        _id: "$name",
        name: { $first: "$name" },
        amountInStock: { $first: "$amountInStock" },
    },
```

Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:263]:
```suggestion
            $group: {
            _id: "$name",
            name: { $first: "$name" },
            amountInStock: { $first: "$amountInStock" },
            },
```

Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:263]:
```suggestion
        { $sort: { amountInStock: bool } }
```

Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:263]:
```suggestion
        { $sort: { amountInStock: 1 } }]);
```

Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:263]:
```suggestion
{ $sort: { amountInStock: 1 } }
```

Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:263]:
```suggestion
{ $sort: { amountInStock: bool } }
```

Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:263]:
```suggestion
        { $sort: { amountInStock: bool } }]);
```

File: backend/resources/product/products.controller.ts (hunk 1 above)
Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:221]:
```suggestion
    { $sort: { amountInStock: bool } }
```

File: backend/resources/product/products.controller.ts [hunk 4]
```diff
@@ -199,4 +199,65 @@ export async function getManufacturers(req, res) {
...
+        },
```
Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:216]:
```suggestion
        $group: {
          _id: "$_id",
          name: { $first: "$name" },
          amountInStock: { $first: "$amountInStock" },
        },
```

File: backend/resources/product/products.controller.ts [hunk 5]
```diff
@@ -199,4 +199,65 @@ export async function getManufacturers(req, res) {
...
+    ]);
+
+      const FilteredStockValue = result.map((Result) => ({
+        Product: Result.name,
+        totalAmount: Result.amountInStock,
+      }));
+      
+      res.status(200).json(FilteredStockValue);
+    
+  } catch (err) {
```
Comment (JohanTran02) [inline backend/resources/product/products.controller.ts:231]: Mappen uppnår samma sak som aggregation #L211-218

**Review by JohanTran02 — CHANGES_REQUESTED** [review]
Only really tiny changes. Dont forget to pull from main before committing or merging. Otherwise very nice code!!!!

File: backend/resources/product/products.controller.ts (hunk 4 above)
Comment (GaKa00) [reply backend/resources/product/products.controller.ts:216]: applied suggestion

**Review by JohanTran02 — APPROVED** [review]
Funkar!

**Pull request merged**